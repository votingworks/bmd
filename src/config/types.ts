// Generic
export interface Dictionary<T> {
  [key: string]: T | undefined
}

// Events
export type InputEvent = React.FormEvent<EventTarget>
export type ButtonEvent = React.MouseEvent<HTMLButtonElement>
export type ButtonEventFunction = (event: ButtonEvent) => void

// Candidates
export interface Candidate {
  readonly id: string
  readonly name: string
  readonly partyId?: string
  isWriteIn?: boolean
}
export type OptionalCandidate = Candidate | undefined

// Contests
export type ContestTypes = 'candidate' | 'yesno'
export interface Contest {
  readonly id: string
  readonly districtId: string
  readonly partyId?: string
  readonly section: string
  readonly title: string
  readonly type: ContestTypes
}
export interface CandidateContest extends Contest {
  readonly type: 'candidate'
  readonly seats: number
  readonly candidates: Candidate[]
  readonly allowWriteIns: boolean
}
export interface YesNoContest extends Contest {
  readonly type: 'yesno'
  readonly description: string
  readonly shortTitle: string
}
export type Contests = (CandidateContest | YesNoContest)[]

// Election
export interface BMDConfig {
  readonly requireActivation?: boolean
  readonly showHelpPage?: boolean
  readonly showSettingsPage?: boolean
}
export interface ElectionDefaults {
  readonly bmdConfig: BMDConfig
}
export interface BallotStyle {
  readonly id: string
  readonly precincts: string[]
  readonly districts: string[]
  readonly partyId?: string
}
export interface Party {
  readonly id: string
  readonly name: string
  readonly abbrev: string
}
export type Parties = Party[]
export interface Precinct {
  readonly id: string
  readonly name: string
}
export interface District {
  readonly id: string
  readonly name: string
}
export interface County {
  readonly id: string
  readonly name: string
}

export interface Election {
  readonly ballotStyles: BallotStyle[]
  readonly county: County
  readonly parties: Parties
  readonly precincts: Precinct[]
  readonly districts: District[]
  readonly contests: Contests
  readonly date: string
  readonly seal?: string
  readonly sealURL?: string
  readonly state: string
  readonly title: string
  readonly bmdConfig?: BMDConfig
}
export type OptionalElection = Election | undefined

export interface ActivationData {
  ballotStyle: BallotStyle
  precinct: Precinct
}

// Votes
export type CandidateVote = Candidate[]
export type YesNoVote = 'yes' | 'no'
export type OptionalYesNoVote = YesNoVote | undefined
export type Vote = CandidateVote | YesNoVote
export type OptionalVote = Vote | undefined
export type VotesDict = Dictionary<Vote>

// Ballot
export type UpdateVoteFunction = (contestId: string, vote: OptionalVote) => void
export type MarkVoterCardUsedFunction = (props: {
  ballotPrinted: boolean
}) => Promise<boolean>
export interface BallotContextInterface {
  activateBallot: (activationData: ActivationData) => void
  ballotStyleId: string
  contests: Contests
  readonly election: Election | undefined
  incrementBallotsPrintedCount: () => void
  isLiveMode: boolean
  markVoterCardUsed: MarkVoterCardUsedFunction
  precinctId: string
  resetBallot: (path?: string) => void
  setUserSettings: (partial: PartialUserSettings) => void
  updateVote: UpdateVoteFunction
  userSettings: UserSettings
  votes: VotesDict
}

// Smart Card Content
export type CardDataTypes = 'voter' | 'pollworker' | 'clerk'
export interface CardData {
  readonly t: CardDataTypes
}
export interface VoterCardData extends CardData {
  readonly t: 'voter'
  readonly bs: string
  readonly pr: string
  readonly uz?: number
  readonly bp?: number
}
export interface PollworkerCardData extends CardData {
  readonly t: 'pollworker'
  readonly h: string
}
export interface ClerkCardData extends CardData {
  readonly t: 'clerk'
  readonly h: string
}

// User Interface
export type ScrollDirections = 'up' | 'down'
export interface ScrollShadows {
  showBottomShadow: boolean
  showTopShadow: boolean
}
export interface Scrollable {
  isScrollable: boolean
}

export type TextSizeSetting = 0 | 1 | 2 | 3

export interface UserSettings {
  textSize: TextSizeSetting
}
export type PartialUserSettings = Partial<UserSettings>

export default {}
