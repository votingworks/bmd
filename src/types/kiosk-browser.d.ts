declare namespace KioskBrowser {
  export interface BatteryInfo {
    discharging: boolean
    level: number // Number between 0â€“1
  }

  export interface PrinterInfo {
    // Docs: http://electronjs.org/docs/api/structures/printer-info
    description: string
    isDefault: boolean
    name: string
    status: number
    // Added via kiosk-browser
    connected: boolean
    options?: { [key: string]: string }
  }

  export interface Device {
    locationId: number
    vendorId: number
    productId: number
    deviceName: string
    manufacturer: string
    serialNumber: string
    deviceAddress: number
  }

  export interface Kiosk {
    print(): Promise<void>
    getPrinterInfo(): Promise<PrinterInfo[]>
    getBatteryInfo(): Promise<BatteryInfo>
    devices: import('rxjs').Observable<Iterable<Device>>
    quit(): void
  }
}

// Disable `no-var` because using `var` ensures `kiosk` is a property on
// `globalThis`, which makes it available both as plain `kiosk` and as
// `window.kiosk`. Using `const` or `let`, as eslint suggests, will not make it
// available at all. An alternative would be to add `kiosk` as a property to
// the `Window` interface, but then we couldn't refer to `kiosk` without
// `window`.
// eslint-disable-next-line no-var
declare var kiosk: KioskBrowser.Kiosk | undefined
