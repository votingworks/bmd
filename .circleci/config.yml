version: 2
jobs:
  test-frontend-and-server:
    docker:
      - image: circleci/python:3.7.3-node-browsers
    steps:
      - checkout
      - restore_cache: # the version on this cache changes rarely, only when a significant setting breaks the cache
          key: dotcache-cache-v20190430-{{ checksum "package.json" }}
      - run:
          name: Setup Dependencies
          command: yarn install --frozen-lockfile
      - run:
          name: Setup Code Climate test-reporter
          command: |
            curl -L https://codeclimate.com/downloads/test-reporter/test-reporter-latest-linux-amd64 > ./cc-test-reporter
            chmod +x ./cc-test-reporter
      - save_cache: # special step to save the dependency cache
          key: dotcache-cache-v20190430-{{ checksum "package.json" }}
          paths:
            - ~/.cache
      - run: # run tests
          name: Run Front-End Test and Coverage
          command: |
            ./cc-test-reporter before-build
            yarn test
            ./cc-test-reporter after-build --exit-code $?
      - run:
          name: Set up python tools for backend
          command: |
            cd server
            pip install pipenv
            pipenv install
      - run:
          name: Run Python Tests and Coverage
          command: |
            cd server
            pipenv run make coverage

workflows:
  version: 2
  test-frontend-and-server:
    jobs:
      - test-frontend-and-server
