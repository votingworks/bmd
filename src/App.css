/* Allow selectors to be used multiple times in this file for grouping of concerns. */
/* stylelint-disable no-duplicate-selectors */

/* Reset */
html {
  box-sizing: border-box;
  line-height: 1;
  user-select: none;
}
*,
*::before,
*::after {
  box-sizing: inherit;
}

body {
  margin: 0;
}

table {
  border-collapse: collapse;
}

fieldset {
  margin: 0;
  border: none;
  padding: 0;
}

legend {
  display: block;
}

:link,
:visited {
  color: rgb(0, 0, 238);
}

:focus {
  outline: rgb(77, 144, 254) dashed 0.25rem;
}

/* Foundational */
html,
body,
#root {
  height: 100%;
  overflow: hidden;
}
@media print {
  html,
  body,
  #root {
    height: auto;
    overflow: visible;
  }
}

/* Typography */
html {
  background: #edeff0;
  color: #263238;
  font-family: -apple-system, BlinkMacSystemFont, 'Segoe UI', 'Roboto', 'Oxygen',
    'Ubuntu', 'Cantarell', 'Fira Sans', 'Droid Sans', 'Helvetica Neue',
    sans-serif;
  font-size: 24px; /* Magic number, must synced with JavaScript global defaults */
  -webkit-font-smoothing: antialiased;
  -moz-osx-font-smoothing: grayscale;
}
@media print {
  html {
    background: #ffffff;

    /* Adjust printed ballot font-size */
    font-size: 16px !important; /* stylelint-disable-line declaration-no-important */
  }
}

/* Accessibility Utilities */
.visually-hidden {
  position: absolute !important; /* stylelint-disable-line declaration-no-important */
  width: 1px;
  height: 1px;
  overflow: hidden;
  clip: rect(1px, 1px, 1px, 1px);
}
.visually-hidden a:focus,
.visually-hidden input:focus,
.visually-hidden button:focus {
  position: static;
  width: auto;
  height: auto;
}

/* Utilities */
.print-only {
  display: none;
}
@media print {
  .print-only {
    display: block;
  }
  .no-print {
    display: none;
  }
}

@media print {
  /* Chrome on Linux prints scrollbars */
  ::-webkit-scrollbar {
    display: none;
  }
}

/* stylelint-disable-next-line selector-class-pattern */
html body .cvox_indicator_container {
  /* Extra specificity/!important ensure this rule is followed. */
  display: none !important; /* stylelint-disable-line declaration-no-important */
}

/* Adjust printed ballot paper settings */
@page {
  margin: 0.375in;
  size: letter portrait;
}
