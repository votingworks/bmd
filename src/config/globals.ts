export default {
  CARD_POLLING_INTERVAL: 200,
  CHECK_ICON: 'âœ“',
  FONT_SIZES: [18, 24, 28, 32],
  TEXT_SIZE: 1,
  YES_NO_VOTES: {
    no: 'No',
    yes: 'Yes',
  },
}
