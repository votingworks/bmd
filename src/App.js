import React, { Component } from 'react';
import PrintProvider, { Print, NoPrint } from 'react-easy-print';
import './App.css';


import QrReader from 'react-qr-scanner';

import Ballot from './Ballot';
import PrintableBallot from './PrintableBallot';

var ELECTION = {
  id: "DEMO",
  title: "Demo Election",
  contests: [
    {
      id: "president",
      type: "plurality",
      name: "President",
      options: [
        "Minnie Mouse",
        "Mickey Mouse",
        "Donald Duck"
        ]
    },
    {
      id: "senator",
      type: "plurality",
      name: "Senator",
      options: [
        "Chad Hanging",
        "Lev Ermachine",
        "John Smith"
        ]
    }
  ]
};

class App extends Component {
  constructor(props) {
    super(props);
    this.state = {};
    this.election = ELECTION;
  }

  ballotReady(ballot) {
    this.setState({ballot: ballot});
  }

  print() {
    window.print();
  }

  handleScan(data) {
    if (data!=null) {
      console.log(data);
    }
  }

  handleError() {
  }

  render() {
    let mainContent, printableBallot;

    if (!this.election) {
      printableBallot = (<div></div>);

      const previewStyle = {
        height: 440,
        width: 620
      };

      mainContent = (
        <div align="center">
          <QrReader
            style={previewStyle}
            onError={this.handleError}
            onScan={this.handleScan}
          />
        </div>
      );
    } else {

      if (this.state.ballot) {
        printableBallot = (
          <PrintableBallot election={this.election} ballot={this.state.ballot} />
        );

        mainContent = (
          <div>
            <div style={{paddingLeft: "400px", paddingRight: "400px", paddingBottom: "50px"}}>{printableBallot}</div>
            <button onClick={this.print.bind(this)} style={{fontSize: "2.5em"}}>Print</button>
          </div>

        );
      } else {
        printableBallot = (<div></div>);
        mainContent = (
          <Ballot election={this.election} onBallotReady={this.ballotReady.bind(this)} />
        );
      }
    }


    return (
      <PrintProvider>
        <Print single printOnly name="ballot">
          {printableBallot}
        </Print>

        <NoPrint force>
          <div className="App">
            <img src="./vw-checkmark.png" className="App-logo" alt="logo" />
            {mainContent}
          </div>
      </NoPrint>
      </PrintProvider>
    );
  }
}

export default App;
