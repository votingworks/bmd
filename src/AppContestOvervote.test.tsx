import React from 'react'
import { fireEvent, render } from 'react-testing-library'

import electionSample from './data/electionSample.json'

import App, { electionKey, mergeWithDefaults } from './App'
import { CandidateContest, Election } from './config/types'

const election = electionSample as Election
const contest0 = election.contests[0] as CandidateContest
const contest0candidate0 = contest0.candidates[0]
const contest0candidate1 = contest0.candidates[1]

const electionSampleAsString = JSON.stringify(
  mergeWithDefaults(electionSample as Election)
)

beforeEach(() => {
  window.localStorage.clear()
  window.location.href = '/'
})

it(`Overvote triggers modal`, () => {
  window.localStorage.setItem(electionKey, electionSampleAsString)
  const { container, getByText, getByTestId } = render(<App />)
  fireEvent.change(getByTestId('activation-code'), {
    target: { value: 'VX.precinct-23.12D' },
  })
  // TODO: replace next line with "Enter" keyDown on activation code input
  fireEvent.click(getByText('Submit'))
  fireEvent.click(getByText('Get Started'))

  // Confirm Correct Contest
  getByText(contest0.title)
  expect(contest0.seats).toEqual(1)

  // Test overvote modal
  fireEvent.click(getByText(contest0candidate0.name).closest('label')!)
  fireEvent.click(getByText(contest0candidate1.name).closest('label')!)
  expect(container.firstChild).toMatchSnapshot()
  getByText(
    `You may only select ${
      contest0.seats
    } candidate in this contest. To vote for ${
      contest0candidate1.name
    }, you must first unselect selected candidate.`
  )

  fireEvent.click(getByText('Okay'))
  expect(
    (getByText(contest0candidate0.name)
      .closest('label')!
      .querySelector('input') as HTMLInputElement).checked
  ).toBeTruthy()
})
