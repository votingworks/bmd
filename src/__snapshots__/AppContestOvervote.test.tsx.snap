// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Overvote triggers modal 1`] = `
.c1 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex: 1;
  -ms-flex: 1;
  flex: 1;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
}

.c1::-webkit-scrollbar {
  display: none;
}

.c3 {
  width: 100%;
  max-width: 66ch;
  line-height: 1.2;
}

.c3 h1 {
  margin: 2rem 0 1rem;
  line-height: 1.1;
  font-size: 1.5rem;
}

.c3 h2 {
  margin: 1.5rem 0 0.75rem;
  font-size: 1.25rem;
}

.c3 h3,
.c3 p {
  margin-top: 1rem;
  margin-bottom: 1rem;
  font-size: 1rem;
}

.c3 h1 + h2 {
  margin-top: -0.75rem;
}

.c3 h1 + p,
.c3 h2 + p {
  margin-top: -0.75rem;
}

.c3 h3 + p {
  margin-top: -1rem;
}

.c3 :first-child {
  margin-top: 0;
}

.c3 :last-child {
  margin-bottom: 0;
}

.c13 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-wrap: wrap-reverse;
  -ms-flex-wrap: wrap-reverse;
  flex-wrap: wrap-reverse;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: justify;
  -webkit-justify-content: space-between;
  -ms-flex-pack: justify;
  justify-content: space-between;
  border-bottom: 1px solid rgb(169,169,169);
  background: #455a64;
  padding: 0.25rem;
}

.c13 > *:first-child {
  -webkit-order: 2;
  -ms-flex-order: 2;
  order: 2;
}

.c13 > * {
  -webkit-flex: 1;
  -ms-flex: 1;
  flex: 1;
  margin: 0.25rem;
}

.c17 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-wrap: wrap-reverse;
  -ms-flex-wrap: wrap-reverse;
  flex-wrap: wrap-reverse;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: justify;
  -webkit-justify-content: space-between;
  -ms-flex-pack: justify;
  justify-content: space-between;
  -webkit-order: -1;
  -ms-flex-order: -1;
  order: -1;
  border-bottom: 1px solid rgb(169,169,169);
  background: #455a64;
  padding: 0.25rem;
}

.c17 > *:first-child {
  -webkit-order: 2;
  -ms-flex-order: 2;
  order: 2;
}

.c17 > * {
  -webkit-flex: 1;
  -ms-flex: 1;
  flex: 1;
  margin: 0.25rem;
}

.c15 {
  border: none;
  border-radius: 0.25rem;
  box-sizing: border-box;
  background: rgb(211,211,211);
  cursor: pointer;
  padding: 0.75rem 1rem;
  line-height: 1.25;
  color: black;
}

.c14 {
  border: none;
  border-radius: 0.25rem;
  box-sizing: border-box;
  background: rgb(71,167,75);
  cursor: pointer;
  padding: 0.75rem 1rem;
  line-height: 1.25;
  color: #FFFFFF;
}

.c11 {
  word-break: break-word;
}

.c2 {
  margin: 0 auto;
  width: 100%;
  max-width: 35rem;
  padding: 1rem 0.75rem 0.5rem;
}

.c4 {
  text-transform: uppercase;
  font-size: 0.85rem;
  font-weight: 600;
}

.c5 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex: 1;
  -ms-flex: 1;
  flex: 1;
  position: relative;
  overflow: auto;
}

.c5::before,
.c5::after {
  position: absolute;
  z-index: 1;
  width: 100%;
  height: 0.25rem;
  content: '';
  -webkit-transition: opacity 0.25s ease;
  transition: opacity 0.25s ease;
}

.c5::before {
  top: 0;
  opacity: 0;
  background: linear-gradient( to bottom, rgb(177,186,190) 0%, transparent 100% );
}

.c5::after {
  bottom: 0;
  opacity: 0;
  background: linear-gradient( to bottom, transparent 0%, rgb(177,186,190) 100% );
}

.c6 {
  -webkit-flex: 1;
  -ms-flex: 1;
  flex: 1;
  overflow: auto;
}

.c7 {
  margin: 0 auto;
  width: 100%;
  max-width: 35rem;
  padding: 0.5rem 0.5rem 1rem;
}

.c8 {
  display: grid;
  grid-auto-rows: minmax(auto,1fr);
  grid-gap: 0.75rem;
}

.c12 {
  display: grid;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  position: relative;
  border-radius: 0.125rem;
  box-shadow: 0 0.125rem 0.125rem 0 rgba(0,0,0,0.14), 0 0.1875rem 0.0625rem -0.125rem rgba(0,0,0,0.12), 0 0.0625rem 0.3125rem 0 rgba(0,0,0,0.2);
  background: #FFFFFF;
  cursor: pointer;
  -webkit-transition: background 0.25s,color 0.25s;
  transition: background 0.25s,color 0.25s;
}

button.c12 {
  text-align: left;
}

.c12:focus-within {
  outline: rgb(77,144,254) dashed 0.25rem;
}

.c12::before {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  position: absolute;
  top: 0;
  bottom: 0;
  left: 0;
  border-right: 1px solid;
  border-color: rgb(211,211,211);
  border-radius: 0.125rem 0 0 0.125rem;
  background: #FFFFFF;
  width: 3rem;
  text-align: center;
  color: #028099;
  font-size: 2rem;
  font-weight: 700;
  content: '';
}

.c12 > div {
  padding: 0.5rem 0.5rem 0.5rem 4rem;
}

.c9 {
  display: grid;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  position: relative;
  border-radius: 0.125rem;
  box-shadow: 0 0.125rem 0.125rem 0 rgba(0,0,0,0.14), 0 0.1875rem 0.0625rem -0.125rem rgba(0,0,0,0.12), 0 0.0625rem 0.3125rem 0 rgba(0,0,0,0.2);
  background: #028099;
  cursor: pointer;
  color: #FFFFFF;
  -webkit-transition: background 0.25s,color 0.25s;
  transition: background 0.25s,color 0.25s;
}

button.c9 {
  text-align: left;
}

.c9:focus-within {
  outline: rgb(77,144,254) dashed 0.25rem;
}

.c9::before {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  position: absolute;
  top: 0;
  bottom: 0;
  left: 0;
  border-right: 1px solid;
  border-color: #028099;
  border-radius: 0.125rem 0 0 0.125rem;
  background: #FFFFFF;
  width: 3rem;
  text-align: center;
  color: #028099;
  font-size: 2rem;
  font-weight: 700;
  content: 'âœ“';
}

.c9 > div {
  padding: 0.5rem 0.5rem 0.5rem 4rem;
}

.c10 {
  margin-right: 0.5rem;
}

.c16 {
  text-align: center;
  color: #FFFFFF;
  -webkit-flex: 2;
  -ms-flex: 2;
  flex: 2;
}

.c0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  height: 100%;
}

.c0:focus {
  outline: none;
}

@media print {
  .c1 {
    -webkit-box-pack: start;
    -webkit-justify-content: flex-start;
    -ms-flex-pack: start;
    justify-content: flex-start;
    padding: 0;
  }
}

@media (min-width:480px) {
  .c3 {
    line-height: 1.3;
  }
}

@media (min-width:480px) {
  .c13 > *:only-child {
    -webkit-flex: 0;
    -ms-flex: 0;
    flex: 0;
    margin: auto;
    min-width: 33.333%;
  }
}

@media print {
  .c13 {
    display: none;
  }
}

@media (min-width:480px) {
  .c17 > *:first-child {
    margin-left: auto;
  }
}

@media (min-width:480px) {
  .c17 > * {
    -webkit-flex: 0;
    -ms-flex: 0;
    flex: 0;
  }
}

@media (min-width:480px) {
  .c17 > *:only-child {
    -webkit-flex: 0;
    -ms-flex: 0;
    flex: 0;
    min-width: 33.333%;
  }
}

@media print {
  .c17 {
    display: none;
  }
}

@media (min-width:720px) {
  .c2 {
    padding: 1rem 1.5rem 0.5rem;
    padding-left: 5rem;
  }
}

@media (min-width:720px) {
  .c7 {
    padding-right: 1rem;
    padding-left: 1rem;
  }
}

@media (min-width:480px) {
  .c12 > div {
    padding: 1rem 1rem 1rem 4rem;
  }
}

@media (min-width:480px) {
  .c9 > div {
    padding: 1rem 1rem 1rem 4rem;
  }
}

@media print {
  .c16 {
    color: #FFFFFF;
  }
}

<div
  class="c0"
  tabindex="-1"
>
  <main
    class="c1"
  >
    <div
      class="c2"
      id="contest-header"
    >
      <div
        class="c3"
        id="audiofocus"
      >
        <h1
          aria-label="President and Vice-President."
        >
          <div
            class="c4"
          >
            United States
          </div>
          President and Vice-President
        </h1>
        <p
          aria-label="Vote for 1. You have selected 1. Use the down arrow to hear your options. Use the right arrow to move to the next contest."
        >
          <strong>
            Vote for 
            1
            .
          </strong>
           You have selected
           
          1
          .
        </p>
      </div>
    </div>
    <div
      class="c5"
    >
      <div
        class="c6"
      >
        <div
          class="c7"
        >
          <div
            class="c8"
          >
            <label
              aria-label="Joseph Barchi and Joseph Hallaren, Federalist."
              class="c9"
              for="barchi-hallaren"
            >
              <input
                class="visually-hidden c10"
                id="barchi-hallaren"
                name="Joseph Barchi and Joseph Hallaren"
                role="option"
                type="checkbox"
                value="barchi-hallaren"
              />
              <div
                class="c3"
              >
                <p
                  class="c11"
                >
                  <strong>
                    Joseph Barchi and Joseph Hallaren
                  </strong>
                  <br />
                  Federalist
                </p>
              </div>
            </label>
            <label
              aria-label="Adam Cramer and Greg Vuocolo, People's."
              class="c12"
              for="cramer-vuocolo"
            >
              <input
                class="visually-hidden c10"
                id="cramer-vuocolo"
                name="Adam Cramer and Greg Vuocolo"
                role="option"
                type="checkbox"
                value="cramer-vuocolo"
              />
              <div
                class="c3"
              >
                <p
                  class="c11"
                >
                  <strong>
                    Adam Cramer and Greg Vuocolo
                  </strong>
                  <br />
                  People's
                </p>
              </div>
            </label>
            <label
              aria-label="Daniel Court and Amy Blumhardt, Liberty."
              class="c12"
              for="court-blumhardt"
            >
              <input
                class="visually-hidden c10"
                id="court-blumhardt"
                name="Daniel Court and Amy Blumhardt"
                role="option"
                type="checkbox"
                value="court-blumhardt"
              />
              <div
                class="c3"
              >
                <p
                  class="c11"
                >
                  <strong>
                    Daniel Court and Amy Blumhardt
                  </strong>
                  <br />
                  Liberty
                </p>
              </div>
            </label>
            <label
              aria-label="Alvin Boone and James Lian, Constitution."
              class="c12"
              for="boone-lian"
            >
              <input
                class="visually-hidden c10"
                id="boone-lian"
                name="Alvin Boone and James Lian"
                role="option"
                type="checkbox"
                value="boone-lian"
              />
              <div
                class="c3"
              >
                <p
                  class="c11"
                >
                  <strong>
                    Alvin Boone and James Lian
                  </strong>
                  <br />
                  Constitution
                </p>
              </div>
            </label>
            <label
              aria-label="Ashley Hildebrand-McDougall and James Garritty, Whig."
              class="c12"
              for="hildebrand-garritty"
            >
              <input
                class="visually-hidden c10"
                id="hildebrand-garritty"
                name="Ashley Hildebrand-McDougall and James Garritty"
                role="option"
                type="checkbox"
                value="hildebrand-garritty"
              />
              <div
                class="c3"
              >
                <p
                  class="c11"
                >
                  <strong>
                    Ashley Hildebrand-McDougall and James Garritty
                  </strong>
                  <br />
                  Whig
                </p>
              </div>
            </label>
            <label
              aria-label="Martin Patterson and Clay Lariviere, Labor."
              class="c12"
              for="patterson-lariviere"
            >
              <input
                class="visually-hidden c10"
                id="patterson-lariviere"
                name="Martin Patterson and Clay Lariviere"
                role="option"
                type="checkbox"
                value="patterson-lariviere"
              />
              <div
                class="c3"
              >
                <p
                  class="c11"
                >
                  <strong>
                    Martin Patterson and Clay Lariviere
                  </strong>
                  <br />
                  Labor
                </p>
              </div>
            </label>
          </div>
        </div>
      </div>
    </div>
  </main>
  <nav
    class="c13"
  >
    <button
      class="c14"
      id="next"
      role="option"
      type="button"
    >
      Next
    </button>
    <button
      class="c15"
      id="previous"
      role="option"
      type="button"
    >
      Back
    </button>
    <p
      class="c16"
    >
      1
       of 
      20
    </p>
  </nav>
  <nav
    class="c17"
  >
    <div />
    <button
      class="c15"
      role="option"
      type="button"
    >
      Help
    </button>
    <button
      class="c15"
      role="option"
      type="button"
    >
      Settings
    </button>
  </nav>
</div>
`;
